File: shoebill_libs/LtrpApi/src/main/java/lt/ltrp/colorpicker/VehicleColorPicker.java
package lt.ltrp.colorpicker;

import lt.ltrp.object.LtrpPlayer;
import net.gtaun.shoebill.constant.VehicleColor;
import net.gtaun.util.event.EventManager;

import java.util.Arrays;

/**
 * @author Bebras
 *         2016.06.14.
 */
public class VehicleColorPicker extends ColorPicker {

    public static AbstractColorPickerBuilder create(LtrpPlayer player, EventManager eventManager) {
        return new VehicleColorPickerBuilder(new VehicleColorPicker(player, eventManager));
    }

    protected VehicleColorPicker(LtrpPlayer player, EventManager eventManager) {
        super(player, eventManager, Arrays.asList(VehicleColor.getColors()));
    }


    private static class VehicleColorPickerBuilder extends AbstractColorPickerBuilder<VehicleColorPicker, VehicleColorPickerBuilder> {

        public VehicleColorPickerBuilder(VehicleColorPicker colorPicker) {
            super(colorPicker);
        }
    }
}

File: shoebill_libs/color-picker/src/main/java/lt/maze/colorpicker/VehicleColorPicker.kt
package lt.maze.colorpicker

import net.gtaun.shoebill.constant.VehicleColor
import net.gtaun.shoebill.entities.Player
import net.gtaun.util.event.EventManager


/**
 * @author Bebras
 * *         2016.06.14.
 */
class VehicleColorPicker protected constructor(player: Player, eventManager: EventManager) :
        ColorPicker(player, eventManager, VehicleColor.get().toList()) {


    class VehicleColorPickerBuilder internal constructor(colorPicker: VehicleColorPicker) :
            ColorPicker.AbstractColorPickerBuilder<VehicleColorPicker, VehicleColorPickerBuilder>(colorPicker)

    companion object {

        fun create(player: Player, eventManager: EventManager): ColorPicker.AbstractColorPickerBuilder<*, *> {
            return VehicleColorPickerBuilder(VehicleColorPicker(player, eventManager))
        }

        fun create(player: Player, eventManager: EventManager, init: VehicleColorPickerBuilder.() -> Unit): VehicleColorPicker {
            val builder = VehicleColorPickerBuilder(VehicleColorPicker(player, eventManager))
            builder.init()
            return builder.build()
        }
    }
}
